test/dotnet-install-tool.Tests/GivenDotnetInstallTool.cs

```diff
@@ -0,0 +1,46 @@
+// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.IO;
+using FluentAssertions;
+using Microsoft.DotNet.Tools.Test.Utilities;
+using Xunit;
+using System.Linq;
+
+namespace Microsoft.DotNet.Cli.Install.Tests
+{
+    public class GivenDotnetInstallTool : TestBase
+    {
+        [Fact]
+        public void ItErrorsWithANonExistentSource()
+        {
+            var result = new InstallCommand()
+                .ExecuteWithCapturedOutput("tool -g --source nonexistent_package_source nonexistent_tool_package");
+
+            result
+                .Should()
+                .Fail()
+                .And
+                .HaveStdOutContaining("nonexistent_package_source");
+        }
+
+        [Fact]
+        public void ItRunsWithTheSpecifiedVerbosity()
```

Comment:
This ok, verbosity depends on nuget and msbuild behavior, to get enough confident, we need to boot up nuget anyway. The assertion is too fragile, what if nuget change behavior. it is a contract nuget verbose:n will print out the package id? What if nuget will echo package id all the time?

`* put on perfectionist hat`
There is test needed:
1. verbosity test in adapter layer. Pass down to nuget as /v:n, or nuget will behave differently.
2. mock test, to cover InstallToolCommand
3. convert --verb to correct method param and help text

This explicit three layer is awkward, maybe we should build something. When parser see `--verbose`, `--source`, it should pass `verbose`, "source feed" etc in "context" like async context, and it should be passed implicit everywhere we call msbuild or nuget. But it should also have a way to prevent other objects modify them along the way.